homeassistant/components/crownstone/entry_manager.py

```diff
@@ -0,0 +1,190 @@
+
+        # A new clientsession is created because the default one does not cleanup on unload
+        self.sse = CrownstoneSSEAsync(
+            email=email,
+            password=password,
+            access_token=self.cloud.access_token,
+            websession=aiohttp_client.async_create_clientsession(self.hass),
+        )
+        # Listen for events in the background, without task tracking
+        asyncio.create_task(self.async_process_events(self.sse))
+        setup_sse_listeners(self)
+
+        # Set up a Crownstone USB only if path exists
+        if self.config_entry.options[CONF_USB_PATH] is not None:
+            await self.async_setup_usb()
+
+        # Save the sphere where the USB is located
+        # Makes HA aware of the Crownstone environment HA is placed in, a user can have multiple
+        self.usb_sphere_id = self.config_entry.options[CONF_USB_SPHERE]
+
+        self.hass.data.setdefault(DOMAIN, {})[self.config_entry.entry_id] = self
+        self.hass.config_entries.async_setup_platforms(self.config_entry, PLATFORMS)
+
+        # HA specific listeners
+        self.config_entry.async_on_unload(
+            self.config_entry.add_update_listener(_async_update_listener)
+        )
+        self.config_entry.async_on_unload(
+            self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self.on_shutdown)
+        )
+
+        return True
+
+    async def async_process_events(self, sse_client: CrownstoneSSEAsync) -> None:
+        """Asynchronous iteration of Crownstone SSE events."""
+        async with sse_client as client:
+            async for event in client:
+                if event is not None:
+                    # Make SSE updates, like ability change, available to the user
+                    self.hass.bus.async_fire(f"{DOMAIN}_{event.type}", event.data)
+
+    async def async_setup_usb(self) -> None:
+        """Attempt setup of a Crownstone usb dongle."""
+        # Trace by-id symlink back to the serial port
+        serial_port = await self.hass.async_add_executor_job(
+            get_port, self.config_entry.options[CONF_USB_PATH]
+        )
+        if serial_port is None:
+            return
+
+        self.uart = CrownstoneUart()
+        # UartException is raised when serial controller fails to open
+        try:
+            await self.uart.initialize_usb(serial_port)
+        except UartException:
+            self.uart = None
+            # Set entry options for usb to null
+            updated_options = self.config_entry.options.copy()
+            updated_options[CONF_USB_PATH] = None
+            updated_options[CONF_USB_SPHERE] = None
+            # Ensure that the user can configure an USB again from options
+            self.hass.config_entries.async_update_entry(
+                self.config_entry, options=updated_options
+            )
+            # Show notification to ensure the user knows the cloud is now used
+            persistent_notification.async_create(
```

Comment:
Would be nice if we could trigger a reconfiguration of the config entry in a config flow instead. Currently we show notifications for reauth flows only.

Reply:
Triggering a new flow is too forced in my opinion. The USB dongle is optional for this integration. If the user wants to try setup again, it can be done from the options. However the user needs to know that the USB setup failed somehow, and we can't assume that the user reads the logs.

Reply:
Persistent notifications should generally not be created be individual integrations. We have standard patterns for notifying the user of a failed integration setup and we don't use persistent notifications in each integration for this.

If the user has selected to use the USB dongle and that fails, it's not too forced to trigger a new flow to fix that problem. Currently though we only support reauth flows natively with frontend notification as far as I know. You can check how it looks if we start a flow with another named step method as source.

Reply:
I have tried that before with:
```python
hass.async_create_task(
   hass.config_entries.flow.async_init(
      DOMAIN,
      context={"source": "usb_config"},
      data={CONF_UNIQUE_ID: entry.entry_id},
   )
)
```
Which did bring up a popup for the integration on the integration page, with a blue border and a "configure" button. It wasn't a very pleasant user experience however because i had to refresh the page first before being able to see it

Reply:
The latter sounds like a frontend issue.